src/app/code/community/Ambimax/PriceImport/Helper/Data.php

```diff
@@ -117,11 +118,172 @@ public function getErpImporter(bool $load = false): Ambimax_PriceImport_Model_Er
             $this->_import = Mage::getModel('ambimax_priceimport/erpImport');
 
             if ($load) {
-                $this->_import->loadCsvData(
-                    Mage::getStoreConfig('ambimax_priceimport/erp_import_options/file_location')
-                );
+                $this->_import->loadCsvData();
             }
         }
         return $this->_import;
     }
+
+    public function getCurrentFile(): array
+    {
+        $fileNames = $this->getFileNames();
+
+        return $this->getNewestFiles($fileNames);
+    }
+
+    public function getFileNames(): array
+    {
+        $connectionInforamtions = $this->_getConnectionInformations();
+
+        //@codingStandardsIgnoreStart
+        $connectionId = ftp_connect($connectionInforamtions['serverAddress']);
+        if (!ftp_login($connectionId, $connectionInforamtions['username'], $connectionInforamtions['password'])) {
+            $this->throwLoginError($connectionInforamtions['serverAddress'], $connectionInforamtions['username']);
+        }
+
+        ftp_pasv($connectionId, true);
+        $fileNames = $this->prepareFileNames(ftp_nlist($connectionId, $connectionInforamtions['path']));
+
+        ftp_close($connectionId);
+        //@codingStandardsIgnoreEnd
+
+        return $fileNames;
+    }
+
+    /**
+     * @return array
+     */
+    protected function _getConnectionInformations(): array
+    {
+        $connectionConfig = array(
+            'username' => $this->getFtpUser(),
+            'password' => $this->getFtpPassword(),
+            'serverAddress' => $this->getFtpServerAddress(),
+            'path' => $this->getFtpFilesLocation(),
+        );
+
+        return $connectionConfig;
+    }
+
+    protected function getFtpUser(): string
+    {
+
+        return Mage::getStoreConfig('ambimax_priceimport/erp_import_options/file_sftp_username');
+
+    }
+
+    protected function getFtpPassword(): string
+    {
+
+        return Mage::getStoreConfig('ambimax_priceimport/erp_import_options/file_sftp_password');
+
+    }
+
+    protected function getFtpServerAddress(): string
+    {
+
+        return Mage::getStoreConfig('ambimax_priceimport/erp_import_options/file_sftp_host');
+
+    }
+
+    protected function getFtpFilesLocation(): string
+    {
+
+        return Mage::getStoreConfig('ambimax_priceimport/erp_import_options/file_sftp_path');
+
+    }
+
+    protected function getFtpFileNames(): string
+    {
+
+        return Mage::getStoreConfig('ambimax_priceimport/erp_import_options/file_sftp_name');
+
+    }
+
+    /**
+     * @param $originalFilenames
+     */
+    public function prepareFileNames($originalFilenames)
+    {
+        if (empty($originalFilenames)) {
+            return;
+        }
+        if (!empty($originalFilenames)) {
+            foreach ($originalFilenames as $name) {
+                $filenames[$name] = substr($name, strripos($name, '/') + 1);
+            }
+        }
```

Comment:
```suggestion
        foreach ($originalFilenames as $name) {
            $filenames[$name] = substr($name, strripos($name, '/') + 1);
        }
```
It is never empty after the first `if`.